metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/adapters/jdbc/JdbcAdapter.java

```diff
@@ -38,6 +39,27 @@
   private JdbcConfig config;
   private String host;
 
+  protected boolean isConnectionClosed() {
+    boolean isClosed = statement == null || connection == null;
+    if(!isClosed) {
+      try {
+        isClosed = statement.isClosed() || connection.isClosed();
+      } catch (SQLException e) {
+        _LOG.error("Unable to maintain open JDBC connection: " + e.getMessage(), e);
+        isClosed = true;
+      }
+    }
+    return isClosed;
+  }
+
+  protected boolean resetConnectionIfNecessary() {
+    if(isConnectionClosed())
+    {
```

Comment:
Sorry, nit-picky, but we have a new line between the if and the '{' which you don't do elsewhere in this code.  Would be nice to stick to one convention.

We need to enhance our code style checks.  Hard to catch and enforce manually.